core/src/main/java/com/datastrato/gravitino/storage/kv/BinaryEntityEncoderUtil.java

```diff
@@ -0,0 +1,185 @@
+  public static final Set<EntityType> VERSION_0_4_0_COMPATIBLE_ENTITY_TYPES =
+      ImmutableSet.of(METALAKE, CATALOG, SCHEMA, TABLE);
+
+  private BinaryEntityEncoderUtil() {}
+
+  /**
+   * Generate the key for name to id mapping. Currently, the mapping is as following.
+   *
+   * <pre>
+   *   Assume we have the following entities:
+   *   metalake: a1        ---- 1
+   *   catalog : a1.b1     ---- 2
+   *   schema  : a1.b1.c   ---- 3
+   *
+   *   metalake: a2        ---- 4
+   *   catalog : a2.b2     ---- 5
+   *   schema  : a2.b2.c   ---- 6
+   *   schema  : a2.b2.c1  ---- 7
+   *
+   *   metalake: a1        ---- 1 means the name of metalake is a1 and the corresponding id is 1
+   * </pre>
+   *
+   * Then we will store the name to id mapping as follows
+   *
+   * <pre>
+   *  a1         -- 1
+   * 	1/b1       -- 2
+   * 	1/2/c      -- 3
+   * 	a2         -- 4
+   * 	4/b2       -- 5
+   * 	4/5/c      -- 6
+   * 	4/5/c1     -- 7
+   * </pre>
+   *
+   * @param nameIdentifier name of a specific entity
+   * @return key that maps to the id of a specific entity. See above, The key maybe like '4/5/c1'
+   * @throws IOException if error occurs
+   */
+  public static String generateKeyForMapping(
+      NameIdentifier nameIdentifier, EntityType entityType, NameMappingService nameMappingService)
+      throws IOException {
+    Namespace namespace = nameIdentifier.namespace();
+    String name = nameIdentifier.name();
+
+    List<EntityType> parentTypes = EntityType.getParentEntityTypes(entityType);
+    long[] ids = new long[namespace.length()];
+    for (int i = 0; i < ids.length; i++) {
+      ids[i] =
+          nameMappingService.getIdByName(
+              concatIdAndName(
+                  ArrayUtils.subarray(ids, 0, i), namespace.level(i), parentTypes.get(i)));
+    }
+
+    return concatIdAndName(ids, name, entityType);
+  }
+
+  /**
+   * Concatenate the namespace ids and the name of the entity. Assuming the namespace ids are [1, 2]
+   * and the name is 'schema', the result will be '1/2/sc_schema'.
+   *
+   * <p>Attention, in order to make this change backward compatible, if the entity type is TABLE, we
+   * will not add a prefix to the name. If the entity type is not TABLE, we will add the prefix to
+   * the name.
+   *
+   * @param namespaceIds namespace ids, which are the ids of the parent entities
+   * @param name name of the entity
+   * @param type type of the entity
+   * @return concatenated string that used in the id-name mapping.
+   */
+  public static String concatIdAndName(long[] namespaceIds, String name, EntityType type) {
+    String context =
+        Joiner.on(NAMESPACE_SEPARATOR)
+            .join(
+                Arrays.stream(namespaceIds).mapToObj(String::valueOf).collect(Collectors.toList()));
+    // We need to make it backward compatible, so we need to check if the name is already prefixed.
+    String mappingName =
+        VERSION_0_4_0_COMPATIBLE_ENTITY_TYPES.contains(type)
```

Comment:
Better to change to `VERSION_0_4_COMPATIBLE_ENTITY_TYPES` to ignore the minor version.

Reply:
@yuqi1129 please fix this issue, thanks!